pkg/resources/marketplace/configmap_catalog_source_reconciler.go

```diff
@@ -68,26 +66,23 @@ func (r *ConfigMapCatalogSourceReconciler) reconcileRegistryConfigMap(ctx contex
 		},
 	}
 
-	err := r.Client.Get(ctx, k8sclient.ObjectKey{Name: configMap.Name, Namespace: configMap.Namespace}, configMap)
-
-	if err != nil && !k8serr.IsNotFound(err) {
-		return "", fmt.Errorf("Failed to get config map %s from %s namespace: %w", configMap.Name, configMap.Namespace, err)
-	} else if k8serr.IsNotFound(err) {
+	or, err := controllerutil.CreateOrUpdate(ctx, r.Client, configMap, func() error {
 		configMap.Data = configMapData
-		if err := r.Client.Create(ctx, configMap); err != nil {
-			return "", fmt.Errorf("Failed to create configmap %s in %s namespace: %w", configMap.Name, configMap.Namespace, err)
-		}
+		return nil
+	})
+	if err != nil {
+		return configMapName, fmt.Errorf("Failed to create/update configmap %s in %s namespace: %w", configMap.Name, configMap.Namespace, err)
+	}
 
+	switch or {
+	case controllerutil.OperationResultCreated:
 		logrus.Infof("Created registry config map for namepsace %s", r.Namespace)
-	} else {
-		if !reflect.DeepEqual(configMap.Data, configMapData) {
-			configMap.Data = configMapData
-			if err := r.Client.Update(ctx, configMap); err != nil {
-				return "", fmt.Errorf("Failed to update configmap %s in %s namespace: %w", configMap.Name, configMap.Namespace, err)
-			}
-
-			logrus.Infof("Updated config map %s in namspace %s", configMapName, r.Namespace)
-		}
+	case controllerutil.OperationResultUpdated:
+		logrus.Infof("Updated config map %s in namspace %s", configMapName, r.Namespace)
+	case controllerutil.OperationResultNone:
+		break
+	default:
+		return configMapName, fmt.Errorf("Unknown controllerutil.OperationResult '%v'", or)
```

Comment:
I would absent from returning an error in the `default`. It's very unlikely to reach this statement, but as long as the value of `err` is `nil` we shouldn't fail, maybe just log it. Anyways, it's very **very** unlikely to that this is going to happen so it's not a big deal.

```suggestion
	default:
		logrus.Infof("Unknown controllerutil.OperationResult '%v'", or)
```

Reply:
Thanks Sergio! That is a very good point! I've applied your suggestion.